### ISSUE TITLE
Tracker Support for motion 

### ISSUE THREAD

**Issue by g-coded** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#364]
Would it be possible to add support for trackers as space drag?  This could be used similar to how Cybershoes work by adding a pressure switch to the tracker and walking in place with a 'drag step'.  Users could also toggle this to allow for room scale movement(or could the driver check for head movement?).  
I know Natural Locomotion does something similar, but this doesn't work with games that don't support analog locomotion.  It also uses more of a 'march step', if a 'drag step was possible we could use a tread/slidemill for movement.

**Comment by Kung-vr** [comment]
Yeah this would be possible. To get it working I'd have to implement support for the GenericTracker device class (right now we just use "TrackedControllerRole_LeftHand" etc) and figure out how bindings and associating the bind to the tracker you want to move should work. Right now we allow explicit binds for example allowing a button on the left controller to activate drag referencing the right hand. I'm not sure but I'm thinking that might get extra complicated if it were meant to support an input through a custom device on the pogo pins of the tracker. I'm thinking also that even if the software side was perfect, it would be physically finicky to get a good experience for the user but I agree it'd be interesting to try.

TLDR: There would need to be some reworking. I'll keep it in mind as a feature for when I'm looking for something to work on but I can't promise it would be soon.

**Comment by g-coded** [comment]
I tried a proof of concept, and it feels convincing enough when just pressing the trigger while my hand is dragging the controller across the floor.  This would require planning where to put a button so that it is activated at the right time though.
I did notice there was some unwanted X-Axis 'jitter' when 'walking' in a straight line, but didn't want to disable that axis for possible strafing(possible filter?).  I also locked the Y-Axis, as it introduced too much 'head bobbing', but I think finding the right gravity setting could fix this, as no Y movement feels like a slide rather than a walk.
